### PR TITLE
Fix the MvNormal distribution and add reparametrize to GP

### PR DESCRIPTION
**Pull request by tirthasheshpatel**
This is a quick rework of #269. This or fixes the MvNormal distribution to not reparametrize the normal distribution as ``MvNormalCholesky``. I have also added reparametrize keyword in GP to let the user choose the underlying distribution of the GP.

### CODE REVIEW

**Comment by tirthasheshpatel** [comment]
Requesting @AlexAndorra for review

File: pymc4/gp/gp.py [hunk 1]
```diff
@@ -138,9 +136,11 @@ def _build_conditional(
...
         # last dimension that we added earlier.
-        return tf.squeeze(mu, axis=[-1]), stabilize(cov, shift=1e-4)
+        return tf.squeeze(mu, axis=[-1]), stabilize(cov)
 
-    def prior(self, name: NameType, X: ArrayLike, **kwargs) -> ContinuousDistribution:
+    def prior(
+        self, name: NameType, X: ArrayLike, *, reparametrize=True, **kwargs
+    ) -> ContinuousDistribution:
         r"""
         Evaluate the GP prior distribution evaluated over the input locations `X`.
```
Comment (AlexAndorra) [inline pymc4/gp/gp.py:145]:
```suggestion
        Evaluate the GP prior distribution over the input locations `X`.
```

File: pymc4/distributions/multivariate.py [hunk 2]
```diff
@@ -348,21 +346,21 @@ class MvNormalCholesky(ContinuousDistribution):
...
-    scale_tril : array
+    scale_tril : array_like
         Lower triangular matrix, such that scale @ scale.T is positive
         semi-definite
 
     Examples
     --------
     Define a multivariate normal variable for a given covariance
-    matrix::
+    matrix.
```
[the remark this answers is not in the document]
Comment (tirthasheshpatel) [reply pymc4/distributions/multivariate.py]: Done!

File: pymc4/distributions/multivariate.py [hunk 3]
```diff
@@ -348,21 +346,21 @@ class MvNormalCholesky(ContinuousDistribution):
...
 
-        covariance_matrix = np.array([[1., 0.5], [0.5, 2]])
-        chol_factor = np.linalg.cholesky(covariance_matrix)
-        mu = np.zeros(2)
-        vals = pm.MvNormalCholesky('vals', loc=loc, scale=chol_factor)
+    >>> covariance_matrix = np.array([[1., 0.5], [0.5, 2]])
+    >>> chol_factor = np.linalg.cholesky(covariance_matrix)
+    >>> mu = np.zeros(2)
+    >>> vals = pm.MvNormalCholesky('vals', loc=loc, scale=chol_factor)
```
[the remark this answers is not in the document]
Comment (tirthasheshpatel) [reply pymc4/distributions/multivariate.py]: You are right! changed it!

**Comment by tirthasheshpatel** [comment]
can we merge this now @AlexAndorra ?

**Comment by AlexAndorra** [comment]
Thanks for this (and for the reminder) @tirthasheshpatel 😉

**Pull request merged**